### PR TITLE
Switch to use knative.dev/pkg instead of github.com/knative/pkg

### PR DESCRIPTION
**Pull request by houshengbo**
In order to keep consistent with all the knative projects, we replace package github.com/knative/pkg with knative.dev/pkg.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -91,6 +95,7 @@ required = [
   name = "github.com/knative/test-infra"
   non-go = false
 
-[[constraint]]
-  branch = "master"
-  name = "github.com/knative/pkg"
+[[prune.project]]
+  name = "knative.dev/pkg"
+  unused-packages = false
+  non-go = false
```
Comment (mattmoor-sockpuppet) [inline Gopkg.toml]:
```suggestion
  non-go = false

```

File: Gopkg.toml [hunk 2]
```diff
@@ -91,6 +95,6 @@ required = [
   name = "github.com/knative/test-infra"
   non-go = false
 
-[[constraint]]
-  branch = "master"
-  name = "github.com/knative/pkg"
+[[prune.project]]
```
Comment (greghaynes) [inline Gopkg.toml]: Why do we need to add this prune?

Comment (houshengbo) [reply Gopkg.toml]: We only keep the packages when they are used.

Comment (houshengbo) [reply Gopkg.toml]: In general, we prune go-tests, non-go and unused-packages. For knative.dev/pkg, we keep non-go, e.g. the license.

Comment (greghaynes) [reply Gopkg.toml]: sounds good

**Comment by greghaynes** [comment]
When I run hack/update-deps.sh with this change I get the following error:

```bash
grouped write of manifest, lock and vendor: failed to export knative.dev/pkg: fatal: failed to unpack tree object 7b4cf0bfe593195857597c822a1fbce8ce198dea
: exit status 128
```

Do we also need to update Gopkg.lock or?

**Comment by houshengbo** [comment]
Just updated this PR. Could you try one more time? @greghaynes

**Comment by greghaynes** [comment]
Works, thanks!

/lgtm
/approve

**Pull request merged**